python/interpret_community/widget/explanation_dashboard.py

```diff
@@ -135,10 +136,6 @@ def _local_port_available(ip, port, rais=True):
                 sock.bind((ip, port))
                 sock.listen(backlog)
                 sock.close()
-                sock = socket.socket(socket.AF_INET6, socket.SOCK_STREAM)
```

Comment:
this current method checks for TCP/IPv4 or TCP/IPv6, the change here would only work on IPv4, not sure why we'd want that. All for keeping the static method, and in the coming months, we'd love to get this and all local server networking  into a separate package, since its not central to interpreting machine learning :)

Reply:
tagging @scottcode for review comment

Reply:
Sorry, just read the issue this is resolving :) thanks for addressing it!

Reply:
discussed with Brandon, based on this issue it sounds like it is already checking ipv6:
https://github.com/interpretml/interpret-community/issues/227

Reply:
I think the key is that we shouldn't be checking both IPv4 and IPv6, only whatever the intended host/ip is. My server doesn't allow binding a socket to IPv6, 

I just realized that my fix wouldn't work if someone explicitly tries to specify an IPv6 address (even if their system doesn't restrict it). 


I coded up a new version and some tests. What do you think? If you like it, should I make a separate pull request or update this one? @rihorn2 @imatiach-msft 

```python
import ipaddress
import logging
import socket

import pytest

logger = logging.getLogger(__name__)

address_families = {
    ipaddress.IPv4Address: socket.AF_INET,
    ipaddress.IPv6Address: socket.AF_INET6,
}


def get_address_family(address):
    try:
        addr_type = type(ipaddress.ip_address(address))
    except ValueError:
        addr_type = None
    return address_families.get(addr_type, socket.AF_INET)


def local_port_available(ip, port, rais=True):
    """
    Borrowed from:
    https://stackoverflow.com/questions/19196105/how-to-check-if-a-network-port-is-open-on-linux
    """
    addr_family = get_address_family(ip)
    try:
        backlog = 5
        sock = socket.socket(addr_family, socket.SOCK_STREAM)
        sock.bind((ip, port))
        sock.listen(backlog)
        sock.close()
    except socket.error as exc:  # pragma: no cover
        if rais:
            error_message = """Port {0} is not available.
            Please specify another port for use via the 'port' parameter"""
            raise RuntimeError(
                error_message.format(port)
            )
        else:
            logger.warning(str(exc))
            return False
    return True


def test_should_be_available():
    port = 5000
    for ip in ('127.0.0.1', 'localhost', '::1'):
        avail = local_port_available(ip, 5000, rais=False)
        logger.info("{}:{} {}".format(repr(ip), port, avail))
        assert avail


def test_not_available():
    backlog = 5
    host = '127.0.0.1'
    port = 5001
    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    sock.bind((host, port))
    sock.listen(backlog)
    with pytest.raises(RuntimeError) as excinfo:
        local_port_available(host, port, rais=True)
    sock.close()
```

Reply:
@scottcode this PR was already merged.  It would be great if you could create a new one, thanks for looking into this!